File: ScriptService/Dto/Sense/MethodInfo.cs
using ScriptService.Extensions;

namespace ScriptService.Dto.Sense {

    /// <summary>
    /// info about a method
    /// </summary>
    public class MethodInfo : SenseEntry {

        /// <summary>
        /// method parameters
        /// </summary>
        public ParameterInfo[] Parameters { get; set; }

        /// <summary>
        /// type returned by method
        /// </summary>
        public string Returns { get; set; }

        /// <inheritdoc />
        public override string ToString() {
            if(!string.IsNullOrEmpty(Returns))
                return $"{Name}({string.Join<ParameterInfo>(",", Parameters)}) : {Returns.GetPlainTypeName()}";
            return $"{Name}({string.Join<ParameterInfo>(",", Parameters)}) : void";
        }
    }
}

File: nc-scripts/src/app/dto/sense/methodinfo.ts
import { SenseEntry } from './senseentry';
import { ParameterInfo } from './parameterinfo';

/**
 * info about a method
 */
export interface MethodInfo extends SenseEntry {

    /**
     * parameters of method
     */
    parameters: ParameterInfo[]

    /**
     * returned type
     */
    returns?: string
}